From:	Tomasz Majchrzak <tomasz.majchrzak@intel.com>
Subject: [PATCH] raid10: increment write counter after bio is split
Date:	Thu, 28 Jul 2016 10:28:25 +0200
Patch-mainline: Not yet, will be upstreamed soon
Reference: bsc#992782

md pending write counter must be incremented after bio is split,
otherwise it gets decremented too many times in end bio callback and
becomes negative.

Signed-off-by: Tomasz Majchrzak <tomasz.majchrzak@intel.com>
Reviewed-by: Artur Paszkiewicz <artur.paszkiewicz@intel.com>
Signed-off-by: Coly Li <colyli@suse.de>
---
 drivers/md/raid10.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/md/raid10.c
+++ b/drivers/md/raid10.c
@@ -1140,6 +1140,8 @@ static void __make_request(struct mddev
 	int max_sectors;
 	int sectors;
 
+	md_write_start(mddev, bio);
+
 	/*
 	 * Register the new request and wait if the reconstruction
 	 * thread has put up a bar for new requests.
@@ -1541,8 +1543,6 @@ static void raid10_make_request(struct m
 		return;
 	}
 
-	md_write_start(mddev, bio);
-
 	do {
 
 		/*
